src/systems/GameSystem.hpp

```diff
@@ -38,6 +46,9 @@ namespace indie
 
     private:
         std::unique_ptr<IScene> createScene();
+        std::shared_ptr<IEntity> createCamera(Vector3 camPos, Vector3 camTarget);
+        static void generateMap(const std::string &filename, IScene &scene);
```

Comment:
Add a -> TODO: trasnform method to none static to avoid forwarding the scene.
Will be done at a later time...